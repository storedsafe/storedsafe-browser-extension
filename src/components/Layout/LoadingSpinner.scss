@import '../../variables';

.loading-spinner {
  position: relative;
  height: 3em;
  width: 3em;
  margin: auto;
}

.spinner {
  &::after,
  &::before {
    border: 0.2em solid transparent;
    border-radius: 100%;
    border-top-color: $color-accent-0;
    box-sizing: border-box;
    content: '';
    height: 100%;
    left: 0;
    position: absolute;
    top: 0;
    width: 100%;
  }

  &::before {
    animation: spin 1s infinite;
    transition: all $transition-duration*3;
    z-index: 100;
  }

  &::after {
    background-image: url('../../../graphics/padlock.svg');
    background-size: contain;
    border-color: $color-primary-1;
    transition: all $transition-duration*3;
  }

  &.error::before,
  &.warning::before,
  &.success::before {
    animation: none;
    border: 0;
  }

  &.error::after {
    background-image: url('../../../graphics/padlock-error.svg');
    border-color: $color-error-0;
  }

  &.warning::after {
    background-image: url('../../../graphics/padlock-warning.svg');
    border-color: $color-warning-0;
  }

  &.success::after {
    background-image: url('../../../graphics/padlock-unlocked.svg');
    border-color: $color-accent-0;
  }
}

@keyframes spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}
