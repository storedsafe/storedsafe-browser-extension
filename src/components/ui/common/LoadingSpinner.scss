@import '../../../variables';

.loading-spinner {
  $size: 3em;

  position: relative;
  height: $size;
  width: $size;
  margin: auto;

  .spinner {
    &::after,
    &::before {
      position: absolute;
      content: '';
      border-radius: 100%;
      box-sizing: border-box;
      left: 0;
      top: 0;
      height: 100%;
      width: 100%;
    }

    &::after {
      border: 0.3em solid #fff;
      border-top-color: $color-accent-0;
      animation: spin 1s infinite;
      transition: all $transition-duration*3;
      z-index: 100;
    }

    &::before {
      background-image: url('../../../graphics/padlock.svg');
      background-size: contain;
      border: 0;
      transition: all $transition-duration*3;
    }

    &.error::after,
    &.warning::after,
    &.success::after {
      animation: none;
      border: 0;
    }

    &.error::before {
      background-image: url('../../../graphics/padlock-error.svg');
      border-color: $color-danger-0;
    }

    &.warning::before {
      background-image: url('../../../graphics/padlock-warning.svg');
      border-color: $color-warning-0;
    }

    &.success::before {
      background-image: url('../../../graphics/padlock-unlocked.svg');
      border-color: $color-accent-0;
    }
  }
}

@keyframes spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}
