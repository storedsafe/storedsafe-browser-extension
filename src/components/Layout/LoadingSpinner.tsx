import React from 'react';
import './LoadingSpinner.scss';

interface LoadingSpinnerProps {
  status?: 'loading' | 'success' | 'error' | 'warning';
}

export const LoadingSpinner: React.SFC<LoadingSpinnerProps> = ({
  status,
}: LoadingSpinnerProps) => (
  <div className="loading-spinner">
    <div className={`spinner ${status}`} />
  </div>
);

LoadingSpinner.defaultProps = {
  status: 'loading',
};
