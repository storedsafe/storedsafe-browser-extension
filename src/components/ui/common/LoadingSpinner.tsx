import React from 'react';
import './LoadingSpinner.scss';

interface LoadingSpinnerProps {
  status?: 'loading' | 'success' | 'error' | 'warning';
}

export const LoadingSpinner: React.FunctionComponent<LoadingSpinnerProps> = ({
  status,
}: LoadingSpinnerProps) => (
  <div className="loading-spinner">
    <div className={`spinner ${status}`} />
  </div>
);

LoadingSpinner.defaultProps = {
  status: 'loading',
};
