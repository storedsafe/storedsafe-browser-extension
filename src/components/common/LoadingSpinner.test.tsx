import * as React from 'react'
import { render, unmountComponentAtNode } from 'react-dom'
import { act } from 'react-dom/test-utils'
import { LoadingSpinner } from './LoadingSpinner'
import pretty from 'pretty'

/**
 * Set up and tear down container to mount tested component in.
 * */
let container: Element = null
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div')
  document.body.appendChild(container)
})
afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container)
  container.remove()
  container = null
})

test('<LoadingSpinner />', () => {
  act(() => {
    render(<LoadingSpinner />, container)
  })
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test('<LoadingSpinner loading />', () => {
  act(() => {
    render(<LoadingSpinner status='loading' />, container)
  })
  expect(container.innerHTML).toMatchSnapshot()
})

test('<LoadingSpinner success />', () => {
  act(() => {
    render(<LoadingSpinner status='success' />, container)
  })
  expect(container.innerHTML).toMatchSnapshot()
})

test('<LoadingSpinner warning />', () => {
  act(() => {
    render(<LoadingSpinner status='warning' />, container)
  })
  expect(container.innerHTML).toMatchSnapshot()
})

test('<LoadingSpinner error />', () => {
  act(() => {
    render(<LoadingSpinner status='error' />, container)
  })
  expect(container.innerHTML).toMatchSnapshot()
})
