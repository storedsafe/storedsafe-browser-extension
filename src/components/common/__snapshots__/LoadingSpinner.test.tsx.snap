// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingSpinner /> 1`] = `
"<div class=\\"loading-spinner\\">
  <div class=\\"spinner loading\\"></div>
</div>"
`;

exports[`<LoadingSpinner error /> 1`] = `"<div class=\\"loading-spinner\\"><div class=\\"spinner error\\"></div></div>"`;

exports[`<LoadingSpinner loading /> 1`] = `"<div class=\\"loading-spinner\\"><div class=\\"spinner loading\\"></div></div>"`;

exports[`<LoadingSpinner success /> 1`] = `"<div class=\\"loading-spinner\\"><div class=\\"spinner success\\"></div></div>"`;

exports[`<LoadingSpinner warning /> 1`] = `"<div class=\\"loading-spinner\\"><div class=\\"spinner warning\\"></div></div>"`;
